File: ralph-workflow/src/cli/args/pause_on_exit.rs
/// Controls whether Ralph pauses for input before process exit.
#[derive(clap::ValueEnum, Debug, Clone, Copy, PartialEq, Eq, Default)]
pub enum PauseOnExitMode {
    /// Pause on failure only when launch appears standalone (double-click Explorer launch).
    #[default]
    Auto,
    /// Always pause before exit, regardless of outcome.
    Always,
    /// Never pause before exit.
    Never,
}

File: ralph-workflow/ralph/config/pause_on_exit.py
"""Pause behavior enum for process exit."""

from enum import StrEnum


class PauseOnExit(StrEnum):
    """Pause behavior before process exit.

    Attributes:
        AUTO: Pause only on standalone failure
        ALWAYS: Always pause before exit
        NEVER: Never pause before exit
    """

    AUTO = "auto"
    ALWAYS = "always"
    NEVER = "never"
